File: Design-Patterns-Using-Cpp/Basics OOPS/Polymorphism.cpp
#include <iostream>

class Adder {
private:
    int sum;

public:
    Adder() : sum(0) {}

    // Overloaded method for incrementing by a specific value
    void increment(int a) {
        this->sum += a;
    }

    // Overloaded method for incrementing by 1
    void increment() {
        this->sum += 1;
    }

    int getValue() const {
        return this->sum;
    }
};

int main() {
    Adder a;
    a.increment(10);
    a.increment();
    std::cout << a.getValue() << std::endl;

    return 0;
}
/*
11
*/

File: Design-Patterns-Using-Ruby/Basics OOPS/Polymorphism.rb
class Adder
    def initialize
      @sum = 0
    end
  
    # Overloaded method for incrementing by a specific value
    def increment(a = 1)
      @sum += a
    end
  
    def get_value
      @sum
    end
end
  
# Client code
a = Adder.new
a.increment(10)
a.increment
puts a.get_value
  